neo/rawio/spikeglxrawio.py

```diff
@@ -272,6 +289,47 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, strea
 
         return raw_signals
 
+    def _event_count(self, event_channel_idx, block_index=None, seg_index=None):
+        timestamps, _, _ = self._get_event_timestamps(block_index, seg_index, event_channel_idx, None, None)
+        return timestamps.size
+
+    def _get_event_timestamps(self, block_index, seg_index, event_channel_index, t_start=None, t_stop=None):
+        timestamps, durations, labels = [], None, []
+        info = self.signals_info_dict[0, "nidq"]  # There are no events that are not in the nidq stream
+        dig_ch = info["digital_channels"]
+        if len(dig_ch) > 0:
+            event_data = self._events_memmap
+            channel = dig_ch[event_channel_index]
+            ch_idx = 7 - int(channel[2:])  # They are in the reverse order
+            this_stream = event_data[:, ch_idx]
+            this_rising = np.where(np.diff(this_stream) == 1)[0] + 1
+            this_falling = (
+                np.where(np.diff(this_stream) == 255)[0] + 1
+            )  # because the data is in unsigned 8 bit, -1 = 255!
+            if len(this_rising) > 0:
+                timestamps.extend(this_rising)
+                labels.extend([f"{channel} ON"] * len(this_rising))
+            if len(this_falling) > 0:
+                timestamps.extend(this_falling)
+                labels.extend([f"{channel} OFF"] * len(this_falling))
+        timestamps = np.asarray(timestamps)
+        if len(labels) == 0:
+            labels = np.asarray(labels, dtype="U1")
+        else:
+            labels = np.asarray(labels)
+        return timestamps, durations, labels
+
+    def _rescale_event_timestamp(self, event_timestamps, dtype, event_channel_index):
+        info = self.signals_info_dict[0, "nidq"]  # There are no events that are not in the nidq stream
+        if not self._use_direct_evt_timestamps:
+            event_times = event_timestamps.astype(dtype) / float(info["sampling_rate"])
+        else:  # Does this ever happen?
+            event_times = event_timestamps.astype(dtype)
+        return event_times
+
+    def _rescale_epoch_duration(self, raw_duration, dtype, event_channel_index):
+        return None
+
```

Comment:
Rescaling event timestamps should always return time so I think the else comment is a good question. The idea is that the get event timestamps should return for whatever the data is in (ie frames, samples, seconds). Then if the data is not in time we use the rescale function to switch it to time. I guess my question is what is the point of using the flag `use_direct_evt_timestamps`?

Reply:
I think as I was writing that piece of code, I did wonder about the necessity of using the flag because spikeglx never actually deals with timestamps on its own, and just sample numbers. So I do think we can just safely say 
```
event_times = event_timestamps.astype(dtype) / float(info["sampling_rate"])
return event_times
```

without any use of conditionals involving that flag!